src/standalone/Vegetation/PlantHydraulics.jl

```diff
@@ -148,32 +149,50 @@ struct PlantHydraulicsParameters{
     conductivity_model::CP
     "Water retention model and parameters"
     retention_model::RP
-    "Root distribution function P(z)"
-    root_distribution::F
+    "Rooting depth parameter (m) - a characteristic depth below which 1/e of the root mass lies"
+    rooting_depth::RD
 end
 
 function PlantHydraulicsParameters(;
     ai_parameterization::PrescribedSiteAreaIndex{FT},
     ν::FT,
     S_s::FT,
-    root_distribution::Function,
+    rooting_depth::Union{Nothing, FT, ClimaCore.Fields.Field} = nothing,
+    root_distribution::Union{Nothing, Function} = nothing,
     conductivity_model,
     retention_model,
 ) where {FT}
-    return PlantHydraulicsParameters{
-        FT,
-        typeof(ai_parameterization),
-        typeof(conductivity_model),
-        typeof(retention_model),
-        typeof(root_distribution),
-    }(
-        ai_parameterization,
-        ν,
-        S_s,
-        conductivity_model,
-        retention_model,
-        root_distribution,
-    )
+    if root_distribution isa Nothing
+        return PlantHydraulicsParameters{
+            FT,
+            typeof(ai_parameterization),
+            typeof(conductivity_model),
+            typeof(retention_model),
+            typeof(rooting_depth),
+        }(
+            ai_parameterization,
+            ν,
+            S_s,
+            conductivity_model,
+            retention_model,
+            rooting_depth,
+        )
+    else
+        return PlantHydraulicsParameters{
+            FT,
+            typeof(ai_parameterization),
+            typeof(conductivity_model),
+            typeof(retention_model),
+            FT,
+        }(
+            ai_parameterization,
+            ν,
+            S_s,
+            conductivity_model,
+            retention_model,
+            FT(1.0 / root_distribution(0.0)),
+        )
+    end
```

Comment:
You are doing manual type dispatch here. Instead, you can just define two methods for the function to return the correct value.

Reply:
Julia doesn't use keyword arguments as dispatch. When I tried defining two methods for the function, the second just overwrote the first. It seems like this was the solution reached here:
https://discourse.julialang.org/t/how-can-kwargs-be-used-in-multiple-dispatch/86790

I could make the keyword arguments positional, but that would make the change breaking. Do you see a better way to do this, or should I revert to making this a breaking change?

Reply:
It's fine! I didn't pay attention to `root_distribution` being a kwarg. Thank you!